code/modules/mob/living/silicon/robot/drone/drone_manufacturer.dm

```diff
@@ -87,6 +87,14 @@
 	var/mob/dead/observer/ghost = user
 	ghost.join_as_drone()
 
+/obj/machinery/drone_fabricator/attack_hand(mob/user)
+	. = ..()
+	if(isdrone(user) && Adjacent(user))
+		if(alert(user, "Would you like to shut down?", null, "Yes", "No") != "Yes")
+			return
+		var/mob/living/silicon/robot/drone/D = user
```

Comment:
```suggestion
		if(alert(user, "Would you like to shut down?", null, "Yes", "No") != "Yes")
			return
		var/mob/living/silicon/robot/drone/D = user
		if(!istype(user))
			return
```
needed in-case the maint drone gets fucked up in between the alert opening and the actual reply being sent.

Reply:
ill throw in a qdeleted too